src/stack.c

```diff
@@ -0,0 +1,54 @@
+#include "stack.h"
+
+stack* init_stack(void) {
+    stack* new = (stack*)malloc(sizeof(stack));
+    new->head = NULL;
+    new->last = NULL;
+    new->size = 0;
+    return new;
+};
+
+node* init_node(element e) {
+    node* new = (node*)malloc(sizeof(node));
+    new->content = e;
+    new->next = NULL;
+    new->previous = NULL;
+    return new;
+};
+
+void add_elem(stack* s, node* n) {
+    if (s->size == 0) {
+        s->head = n;
+        s->last = n;
+        s->size = 1;
+    } else {
+        s->last->next = n;
+        s->last = n;
+        n->previous = s->last;
```

Comment:
It's a loop there :wink:

```suggestion
        n->previous = s->last;
        s->last = n;
```